hermes-api/src/main/java/pl/allegro/tech/hermes/api/SubscriptionMetrics.java

```diff
@@ -67,9 +95,88 @@ public boolean equals(Object obj) {
         return Objects.equals(this.delivered, other.delivered)
             && Objects.equals(this.discarded, other.discarded)
             && Objects.equals(this.rate, other.rate)
+            && Objects.equals(this.httpStatusCodesM1, other.httpStatusCodesM1)
+            && Objects.equals(this.otherErrorsM1, other.otherErrorsM1)
+            && Objects.equals(this.timeoutsM1, other.timeoutsM1)
             && Objects.equals(this.state, other.state);
     }
 
+    private static class HttpStatusCodeMetrics {
+        public String class1xx;
+        public String class2xx;
+        public String class3xx;
+        public String class4xx;
+        public String class5xx;
+
+        public HttpStatusCodeMetrics() {
```

Comment:
i'm not a fan of producing this kind of code for some tools - if you must, use `@JsonCreator`